### PR TITLE
Register auto

### PR DESCRIPTION
**Pull request by AndreasLc1103**

### CODE REVIEW

File: src/com/nutrons/framework/Robot.java [hunk 1]
```diff
@@ -81,6 +87,10 @@ public final void startCompetition() {
   protected final void disabled() {
   }
 
+  public Command registerAuto() {
```
Comment (ashergottlieb) [inline src/com/nutrons/framework/Robot.java]: I believe this should be protected.

File: src/com/nutrons/framework/Robot.java [hunk 2]
```diff
@@ -69,6 +71,10 @@ protected final void robotInit() {
    */
   @Override
   public final void robotMain() {
+    Command auto = this.registerAuto();
+    if (auto != null) {
+      this.competitionStream().filter(x -> x == AUTO).subscribe(x -> auto.terminable(competitionStream().filter(y -> y != AUTO)).execute());
```
Comment (ashergottlieb) [inline src/com/nutrons/framework/Robot.java:76]: Might want to move `auto.terminable(...)` part to go on the `this.registerAuto()` line. Would be easier to read.

**Review by ashergottlieb — APPROVED** [review]
Other than these small changes, I think this should work.

File: src/com/nutrons/framework/Robot.java (hunk 2 above)
Comment (MiloDavis) [reply src/com/nutrons/framework/Robot.java:76]: That would mean that the null check would need to be a ternary operator.  That seems clumsy

File: src/com/nutrons/framework/Robot.java (hunk 1 above)
Comment (MiloDavis) [reply src/com/nutrons/framework/Robot.java]: @AndreasLc1103

Comment (MiloDavis) [reply src/com/nutrons/framework/Robot.java]: @totallyashes The CI failed to build, but when I reran it, things built without incident.  Is this potentially concurrency related?

**Review by MiloDavis — APPROVED** [review]

**Pull request merged**